tests/integration/clientv3/util.go

```diff
@@ -119,3 +125,62 @@ func populateDataIntoCluster(t *testing.T, cluster *integration2.Cluster, numKey
 		}
 	}
 }
+
+// guaranteeRaftAppliedIndexAdvance ensures raftLog.applied is the same as appliedIndex of etcdserver.EtcdServer
+// it should only be used between each raft proposal.
+func guaranteeRaftAppliedIndexAdvance(t *testing.T, gRPCURL string, clientURL url.URL) {
+	t.Logf("ensuring applied index advance, (gRPCURL %s), (clientURL %v)", gRPCURL, clientURL)
+	cc, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{gRPCURL},
+		DialTimeout: 5 * time.Second,
+	})
+	require.NoError(t, err)
+	defer cc.Close()
+	cresp, err := cc.Status(context.TODO(), gRPCURL)
+	require.NoError(t, err)
+	appliedIndex := cresp.RaftAppliedIndex
+
+	require.Eventually(t, func() bool {
+		tr, err := transport.NewTransport(transport.TLSInfo{}, 5*time.Second)
```

Comment:
nit: I think getting raftStatus should be it's own helper method

Reply:
Yeah, I thought guaranteeRaftAppliedIndexAdvance can be reused to solve the similar issue. I will create a separate function for get raft status reuse. Thanks for the suggestion.

Reply:
Updated in the new revision.